ksqldb-rest-model/src/main/java/io/confluent/ksql/rest/entity/SourceDescription.java

```diff
@@ -68,7 +69,9 @@ public SourceDescription(
       @JsonProperty("partitions") final int partitions,
       @JsonProperty("replication") final int replication,
       @JsonProperty("statement") final String statement,
-      @JsonProperty("queryOffsetSummaries") final List<QueryOffsetSummary> queryOffsetSummaries) {
+      @JsonProperty("queryOffsetSummaries") final List<QueryOffsetSummary> queryOffsetSummaries,
+      @JsonProperty("sourceConstraints") final List<String> sourceConstraints
```

Comment:
what happens if we get old response with new CLI and vice versa?

Reply:
From an old CLI to the new Server works.

From the new CLI to the old Server doesn't work. But, is this a valid support? And, how do others maintain compatibility with an old Server? I was looking at the code for other properties and I don't see how this is supported. Any idea?

Reply:
> From an old CLI to the new Server works.

This one is more important. The other is a nice-to-have. It's not expected to work, but it would be nice if it behaved nicely (either ignoring it or telling you there's an issue). You don't need to implement that for this pr